CHANGELOG.md

```diff
@@ -1,4 +1,6 @@
 ## Unreleased changes
+- Align data structures like records, classes and structs to be used uniform throughout the library.
```

Comment:
It would be nice if this changelog was as detailed as the description in the PR, since there are a number of breaking changes.
The purpose is to help people understand what changed and help them migrate their code.

Reply:
Updated with more explanations.